pallets/pallet-bonded-coins/src/tests/transactions/mint_into.rs

```diff
@@ -1 +1,705 @@
+				generate_pool_details(
+					vec![currency_id1],
+					curve.clone(),
+					true,
+					None,
+					None,
+					Some(DEFAULT_COLLATERAL_CURRENCY_ID),
+					None,
+				),
+			),
+			(
+				pool_id2.clone(),
+				generate_pool_details(
+					vec![currency_id2],
+					curve,
+					true,
+					None,
+					None,
+					Some(DEFAULT_COLLATERAL_CURRENCY_ID),
+					None,
+				),
+			),
+		])
+		.build()
+		.execute_with(|| {
+			let origin: OriginFor<Test> = RawOrigin::Signed(ACCOUNT_00).into();
+
+			// pool 1: 1 mint of 10 * amount
+			assert_ok!(BondingPallet::mint_into(
+				origin.clone(),
+				pool_id1.clone(),
+				0,
+				ACCOUNT_00,
+				amount_to_mint * 10,
+				account_collateral / 2,
+				1
+			));
+
+			// pool 2: 10 mints of amount
+			for _ in 0..10 {
+				assert_ok!(BondingPallet::mint_into(
+					origin.clone(),
+					pool_id2.clone(),
+					0,
+					ACCOUNT_00,
+					amount_to_mint,
+					account_collateral / 2 / 10,
+					1
+				));
+			}
+
+			assert_eq!(
+				<Test as crate::Config>::Fungibles::total_balance(currency_id1, &ACCOUNT_00),
+				<Test as crate::Config>::Fungibles::total_balance(currency_id2, &ACCOUNT_00),
+			);
+
+			assert_eq!(
+				<Test as crate::Config>::CollateralCurrencies::total_balance(DEFAULT_COLLATERAL_CURRENCY_ID, &pool_id1),
+				<Test as crate::Config>::CollateralCurrencies::total_balance(DEFAULT_COLLATERAL_CURRENCY_ID, &pool_id2),
+			);
+		})
+}
+
+#[test]
+fn mint_with_frozen_balance() {
+	let pool_id = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+
+	let initial_collateral = 10u128.pow(20);
+	let amount_to_mint = 10u128.pow(10);
+
+	ExtBuilder::default()
+		.with_native_balances(vec![(ACCOUNT_00, ONE_HUNDRED_KILT)])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.with_bonded_balance(vec![(DEFAULT_COLLATERAL_CURRENCY_ID, ACCOUNT_00, initial_collateral)])
+		.with_pools(vec![(
+			pool_id.clone(),
+			generate_pool_details(
+				vec![DEFAULT_BONDED_CURRENCY_ID],
+				get_linear_bonding_curve(),
+				false, // Non-transferable
+				None,
+				None,
+				Some(DEFAULT_COLLATERAL_CURRENCY_ID),
+				None,
+			),
+		)])
+		.build()
+		.execute_with(|| {
+			let origin: OriginFor<Test> = RawOrigin::Signed(ACCOUNT_00).into();
+
+			assert_ok!(BondingPallet::mint_into(
+				origin.clone(),
+				pool_id.clone(),
+				0,
+				ACCOUNT_00,
+				amount_to_mint,
+				initial_collateral,
+				1
+			));
+
+			assert_eq!(
+				<Test as crate::Config>::Fungibles::total_balance(DEFAULT_BONDED_CURRENCY_ID, &ACCOUNT_00),
+				amount_to_mint
+			);
+
+			// Check that balance is frozen
+			assert_eq!(
+				<Test as crate::Config>::Fungibles::reducible_balance(
+					DEFAULT_BONDED_CURRENCY_ID,
+					&ACCOUNT_00,
+					Preservation::Expendable,
+					Fortitude::Polite
+				),
+				0
+			);
+
+			// check that we can mint again into a frozen account
+			assert_ok!(BondingPallet::mint_into(
+				origin,
+				pool_id,
+				0,
+				ACCOUNT_00,
+				amount_to_mint,
+				initial_collateral,
+				1
+			));
```

Comment:
can we also check, if the reducible balance is still zero?

Reply:
2bcc31db8788cd97c6bbeba3d8797e7cf2fc487b